pkg/checks/external/ssl_util/main.go

```diff
@@ -17,37 +17,80 @@ import (
 
 var TimeoutSeconds = 10
 
-// CertHandshake performs a basic TLS/SSL handshake on the specified host/port, returning any errors
-func CertHandshake(host, port string) error {
-	log.Info("Testing SSL handshake on host ", host, " over port ", port)
+// CertificatePuller makes an insecure connection to a host and specified port, and returns the SSL certificate if the DNSNames field matches the hostname
+func CertificatePuller(host, port string) (*x509.Certificate, error) {
+	var hostCert *x509.Certificate
+	d := &net.Dialer{
+		Timeout: time.Duration(TimeoutSeconds) * time.Second,
+	}
+
+	conn, err := tls.DialWithDialer(d, "tcp", host+":"+port, &tls.Config{
+		InsecureSkipVerify: true,
+		MinVersion:         tls.VersionTLS12,
+	})
+
+	if err != nil {
+		log.Warnln([]*x509.Certificate{&x509.Certificate{}}, "", err)
+		return hostCert, err
+	}
```

Comment:
huh so there's potential here to return an empty cert...

Reply:
Updated all functions to throw errors if the self-signed certificates could not be pulled, or could not be appended to the certificate pool.